libnmstate/state.py

```diff
@@ -381,9 +381,21 @@ def merge_interfaces(self, other_state):
         This is a reverse recursive update operation.
         """
         other_state = State(other_state.state)
+        origin_other_state = other_state
+        other_state = State(origin_other_state.state)
         for name in self.interfaces.keys() & other_state.interfaces.keys():
             dict_update(other_state.interfaces[name], self.interfaces[name])
             self._ifaces_state[name] = other_state.interfaces[name]
+        # Merge the port list of linux bridge
+        other_state = State(origin_other_state.state)
+        for iface_name, iface_state in self.interfaces.items():
+            if (
+                iface_state[Interface.STATE] == InterfaceState.UP
+                and iface_state[Interface.TYPE] == LinuxBridge.TYPE
+            ):
+                other_iface_state = other_state.interfaces.get(iface_name)
+                if other_iface_state:
+                    merge_linux_bridge_ports(iface_state, other_iface_state)
```

Comment:
As you are not modifying `other_state` here, you can use `origin_other_state` instead of  `other_state`.

Reply:
Thanks!